scripts/beekeeper.pl

```diff
@@ -507,6 +514,49 @@ sub register_beekeeper {
     return $beekeeper;
 }
 
+
+sub big_red_button {
+  my ( $self, $valley ) = @_;
+
+  my $bk_a = $self->{dba}->get_Beekeeper();
+  my $blocked_beekeepers;
+
+  # Save a list of IDs of beekeepers which were blocked earlier so
+  # that we can not mention them while reporting the current blocking.
+  $blocked_beekeepers = $bk_a->fetch_all( 'is_blocked = 1' );
+  my %previously_blocked_ids;
+  while ( my $blocked_bk = shift @{ $blocked_beekeepers } ) {
+    $previously_blocked_ids{ $blocked_bk->dbID() } = 1;
+  }
+
+  # Begin the shutdown by blocking all registered beekeepers so that
+  # none of them start spawning new workers just as this one tries to
+  # kill all workers.
+  $bk_a->block_all_alive_beekeepers();
+
+  # Report which beekeepers we have just blocked
+  $blocked_beekeepers = $bk_a->fetch_all( 'is_blocked = 1' );
+  my @currently_blocked = grep {
+    ! exists $previously_blocked_ids{ $_->dbID() }
+  } @{ $blocked_beekeepers };
+  while ( my $blocked_bk = shift @currently_blocked ) {
+    # FIXME: decide what else to print
+    printf( "Blocked beekeeper: %10d %35s %35s %15s\n",
+            $blocked_bk->dbID(), $blocked_bk->meadow_host(),
+            $blocked_bk->meadow_user(), $blocked_bk->process_id() );
```

Comment:
All objects but Beekeeper have a `toString` method. Maybe this is the right time to introduce one ?

Reply:
Good idea! There is one now and _big_red_button()_ uses it; might make sense to make it report more, though.

Moreover, _Queen::kill_all_workers()_ now uses _Worker::toString()_ instead of crafting its own summary string.